### DISCUSSION TITLE
Oh My Zsh made my terminal completely unusable. Help please!

### DISCUSSION THREAD

**Discussion by ChaseTheWebDev** [ohmyzsh/ohmyzsh#12269] [Help] [answered]
Hello everyone. I'm new to web-development. I've only been studying for a year now, so this could be user error. I'm not sure.

Yesterday, I created a local mySQL database for a project I'm working on. The only thing terminal-wise I did there was I created a "zprofile" because my iMac did not have one. Inside the .zprofile,  I simply wrote a path for mySQL. I just want to mention I did all this if perhaps it is relevant. 

Now, after I changed the variable, I decided I wanted to make my terminal look better. I have ohmysh on my laptop (installed by my brother, a web dev), so I thought it would be nice on my desktop as well. The default terminal for my iMac is bash (2019), but years ago I changed it (somehow) to permanently use zsh. For years, I have had no issues...  Enter ohmyzsh.

Yesterday after installing oh my zsh I shut my computer down and ended the day. 

Today, my terminal greets me with the following code:

/etc/zshrc:7: command not found: locale
/etc/zshrc_Apple_Terminal:104: command not found: mkdir
/Users/username/.oh-my-zsh/oh-my-zsh.sh:65: command not found: mkdir
/Users/username/.oh-my-zsh/tools/check_for_upgrade.sh:30: command not found: git
/Users/username/.oh-my-zsh/oh-my-zsh.sh:124: command not found: rm
compdump:138: command not found: mv

I want to re-state that I do not press anything for this to appear. I've quit the system multiple times and I can't get it to go away. 
The worst part? Nothing I write beneath this works. Everything returns "Command Not Found" including "Uninstal_oh-my-zsh."

Initially, I thought it was an issue with node, so I uninstalled it and then re-installed it. That didn't work.

I've checked with ChatGPT, and it has no definite answer other than completely rewriting my zshrc file manually, which I am not trained to do. 

Digging around in Finder to avoid using the terminal, here are some things I have found in my users/username file:

An empty (0kb) pre-ohmy-zsh file.
The entire oh-my-zsh directory.
The .zprofile i created (by using "touch" in the command line) and the mySQL PATH i wrote in it.

There are also a bunch of compdumps as well as many unrelated installments like vsCode and Yarn.

I'm really not sure what caused this issue but I have a  STRONG feeling it was oh-my-zsh. 

Can anyone help me? I would be SO grateful.

**Comment by mcornella** [comment]
What did you put in .zprofile? Was it maybe `path=` something?

**Reply by ChaseTheWebDev** [reply]
This is the only thing written in my .zprofile (I simply am copy and pasting it below):
```
export PATH={PATH}:/usr/local/mysql-8.0.36-macos14-x86_64/bin
```
EDIT:
Also, I found a record of all my terminal history. I'm going to copy and paste the inputs I made during which time the error occurred. Maybe you can see something there. (?)
You can see when I downloaded ohmyzsh as well as when I began having troublle with commands (checking node version over and over because I thought there was an error with node).
```
mysql --version
mysql--version
ls /usr/local/mysql
pwd
echo $PATH
/usr/local/mysql-8.0.36-macos14-x86_64/bin/mysql ; exit;
clear
mysql -u root -p
open -t .bash_profile
open -t .zsh_profile
clear
sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
ls -a
touch .zprofile
nano .zprofile
cat .zprofile
mysql -u root -p
clear
mysql -u root -p
clear
quit
npx prisma db push
node -v
node -v 
node -v
uninstall_oh_my_zsh
```


-------------------------

Finally, here is the copy and paste of my entire .zshrc file (I have removed the number signs indicating comments because this markup editor makes it a heading. Everything else is untouched.):
```
 If you come from bash you might have to change your $PATH.
 export PATH=$HOME/bin:/usr/local/bin:$PATH

 Path to your oh-my-zsh installation.
export ZSH="$HOME/.oh-my-zsh"

 Set name of the theme to load --- if set to "random", it will
 load a random theme each time oh-my-zsh is loaded, in which case,
 to know which specific one was loaded, run: echo $RANDOM_THEME
 See https://github.com/ohmyzsh/ohmyzsh/wiki/Themes
ZSH_THEME="robbyrussell"

 Set list of themes to pick from when loading at random
 Setting this variable when ZSH_THEME=random will cause zsh to load
a theme from this variable instead of looking in $ZSH/themes/
 If set to an empty array, this variable will have no effect.
ZSH_THEME_RANDOM_CANDIDATES=( "robbyrussell" "agnoster" )

 Uncomment the following line to use case-sensitive completion.
 CASE_SENSITIVE="true"

 Uncomment the following line to use hyphen-insensitive completion.
 Case-sensitive completion must be off. _ and - will be interchangeable.
 HYPHEN_INSENSITIVE="true"

 Uncomment one of the following lines to change the auto-update behavior
 zstyle ':omz:update' mode disabled  # disable automatic updates
 zstyle ':omz:update' mode auto      # update automatically without asking
 zstyle ':omz:update' mode reminder  # just remind me to update when it's time

 Uncomment the following line to change how often to auto-update (in days).
 zstyle ':omz:update' frequency 13

 Uncomment the following line if pasting URLs and other text is messed up.
 DISABLE_MAGIC_FUNCTIONS="true"

 Uncomment the following line to disable colors in ls.
 DISABLE_LS_COLORS="true"

 Uncomment the following line to disable auto-setting terminal title.
 DISABLE_AUTO_TITLE="true"

Uncomment the following line to enable command auto-correction.
ENABLE_CORRECTION="true"

 Uncomment the following line to display red dots whilst waiting for completion.
 You can also set it to another string to have that shown instead of the default red dots.
 e.g. COMPLETION_WAITING_DOTS="%F{yellow}waiting...%f"
 Caution: this setting can cause issues with multiline prompts in zsh < 5.7.1 (see #5765)
 COMPLETION_WAITING_DOTS="true"

 Uncomment the following line if you want to disable marking untracked files
 under VCS as dirty. This makes repository status check for large repositories
 much, much faster.
 DISABLE_UNTRACKED_FILES_DIRTY="true"

 Uncomment the following line if you want to change the command execution time
 stamp shown in the history command output.
You can set one of the optional three formats:
 "mm/dd/yyyy"|"dd.mm.yyyy"|"yyyy-mm-dd"
 or set a custom format using the strftime function format specifications,
 see 'man strftime' for details.
 HIST_STAMPS="mm/dd/yyyy"

 Would you like to use another custom folder than $ZSH/custom?
 ZSH_CUSTOM=/path/to/new-custom-folder

 Which plugins would you like to load?
 Standard plugins can be found in $ZSH/plugins/
 Custom plugins may be added to $ZSH_CUSTOM/plugins/
 Example format: plugins=(rails git textmate ruby lighthouse)
Add wisely, as too many plugins slow down shell startup.
plugins=(git)

source $ZSH/oh-my-zsh.sh

 User configuration

 export MANPATH="/usr/local/man:$MANPATH"

 You may need to manually set your language environment
 export LANG=en_US.UTF-8

Preferred editor for local and remote sessions
 if [[ -n $SSH_CONNECTION ]]; then
   export EDITOR='vim'
 else
   export EDITOR='mvim'
 fi

 Compilation flags
 export ARCHFLAGS="-arch x86_64"

 Set personal aliases, overriding those provided by oh-my-zsh libs,
 plugins, and themes. Aliases can be placed here, though oh-my-zsh
users are encouraged to define aliases within the ZSH_CUSTOM folder.
For a full list of active aliases, run `alias`.

 Example aliases
 alias zshconfig="mate ~/.zshrc"
 alias ohmyzsh="mate ~/.oh-my-zsh"
```

**Reply by mcornella** [reply, marked as answer]
Hi Chase! If the content of `.zprofile` is a literal copy-paste, the `PATH` variable is missing a dollar, otherwise the previous contents are not preserved. Here's how it should look:

```sh
# double quotes added for good measure, in case there are spaces in some of the paths
export PATH="${PATH}:/usr/local/mysql-8.0.36-macos14-x86_64/bin"
```

As to code formatting in GitHub, it uses Markdown, so you can use [code blocks](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code) so that code is shown as is (and even with syntax highlighting).

**Reply by ChaseTheWebDev** [reply]
You, sir, are a scholar and a saint. Thank you for helping me (and anyone else on here). We need more helpful people like you in the world. Life-saver! 

Also, apologies to OhMyZsh for all the things I cursed it for. I take it back! Terminal looks great again! Woohoo!
